src/solvers/smt2/smt2_conv.cpp

```diff
@@ -2499,6 +2499,14 @@ void smt2_convt::convert_expr(const exprt &expr)
       out << ')';
     }
   }
+  else if(expr.id() == ID_r_ok || expr.id() == ID_w_ok || expr.id() == ID_rw_ok)
+  {
+    convert_expr(simplify_expr(to_r_or_w_ok_expr(expr).lower(ns), ns));
```

Comment:
no coverage here,  could this be a sign that these expressions still get lowered before reaching that step ?

Reply:
This should now be ok - at the time of you viewing this I had not included the removal of early expansion.